nilearn/conftest.py

```diff
@@ -15,6 +15,22 @@
     collect_ignore = ['plotting',
                       'reporting',
                       ]
+    matplotlib = None
+
+
+def pytest_configure(config):
+    """Use Agg so that no figures pop up."""
+    if matplotlib is not None:
+        matplotlib.use('Agg', force=True)
```

Comment:
I hope we will not be having issues as in https://github.com/nilearn/nilearn/issues/2560

Reply:
The expectation for running tests should be different from the expectation when importing a library four actual use. In test running it's fine to set the backend. We've done it this way for years in MNE-Python without issue.